pkg/library/container/mountSources.go

```diff
@@ -84,20 +88,43 @@ func handleMountSources(k8sContainer *corev1.Container, devfileContainer *dw.Con
 		Name:  devfileConstants.ProjectsSourceEnvVar,
 		Value: path.Join(sourceMapping, projectsSourcePath),
 	})
+
+	return nil
 }
 
-// getProjectSourcePath gets the path, relative to PROJECTS_ROOT, that should be used for the PROJECT_SOURCE env var
-func getProjectSourcePath(projects []dw.Project) string {
-	projectPath := ""
+// getProjectSourcePath gets the path, relative to PROJECTS_ROOT, that should be used for the PROJECT_SOURCE env var.
+// Returns an error if there was a problem retrieving the selected starter project.
+//
+// The project source path is determined based on the following priorities:
+//
+//  1. If the workspace has at least one regular project, the first one will be selected.
+//     If the first project has a clone path, it will be used, otherwise the project's name will be used as the project source path.
+//
+//  2. If the workspace has a starter project that is selected, its name will be used as the project source path.
+//
+//  3. If the workspace has at least one starter project, but none are selected,
+//     the name of the first starter project will be used as the project source path.
+//
+//  4. Otherwise, the returned project source path will be an empty string.
+func getProjectSourcePath(workspace *dw.DevWorkspaceTemplateSpec) (string, error) {
+	projects := projectslib.GetAllProjects(workspace)
 	if len(projects) > 0 {
+		selectedStarterProject, err := projectslib.GetStarterProject(workspace)
+		if err != nil {
+			return "", err
+		}
+		// Only use the selected starterProject if we have no regular projects
+		if selectedStarterProject != nil && len(workspace.Projects) == 0 {
+			return selectedStarterProject.Name, nil
+		}
+
 		firstProject := projects[0]
```

Comment:
This relies on `GetAllProjects` returning the projects with starterProjects being in the list after regular projects, which should either be documented as a requirement on the function (and maybe have a test case to verify) or be handled differently.

IMO it's easier to just check `workspace.Projects` first, and if it's empty check `workspace.StarterProjects`.